### PR TITLE
build: Semantic release

### PR DESCRIPTION
**Pull request by felixrieseberg**
In an attempt to make things easier for all of us, I've installed `semantic-release` in this repository. This makes our life easier by automatically generating changelogs and automatically releasing.

It does however require us to use commit conventions. I went with https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-eslint, which doesn't match the current commit format any of us use but seems like it's overall the easiest one 😆 

In other words: Whenever we merge a PR, Travis will automatically push a new release to both GitHub and npm.

cc @jacobq @pichfl @bendemboski

### CODE REVIEW

**Comment by jacobq** [comment]
I was actually thinking about suggesting use of a standard commit message format (with the primary goal of automating change logs). Not to get bogged down in minutia, but I have a few requests:

* Specify commit message convention somewhere that automated tools (I am thinking of [`commitizen`](https://github.com/commitizen/cz-cli) but perhaps there are others) can pick it up.
  `yarn global add commitizen`
  `commitizen init cz-conventional-changelog --yarn --dev --exact`
* Don't ship both `package-lock.json` and `yarn.lock` (just pick one) as usually one gets out of date and becomes misleading (I suppose that's not really related to this PR, but I saw that `yarn.lock` changed so was reminded again)

**Comment by felixrieseberg** [comment]
@jacobq 


> Specify commit message convention somewhere that automated tools (I am thinking of commitizen but perhaps there are others) can pick it up.
    yarn global add commitizen
    commitizen init cz-conventional-changelog --yarn --dev --exact

That's already the case here - I'm using `conventional-changelog`, which is the one supported by `semantic-release`. We've had really good experiences with it over in `electron`.

> Don't ship both package-lock.json and yarn.lock (just pick one) as usually one gets out of date and becomes misleading (I suppose that's not really related to this PR, but I saw that yarn.lock changed so was reminded again)

I didn't want to remove `yarn.lock` but I really do want a `package-lock.json` (after all, the majority of people currently uses `npm`). If you're cool with cutting `yarn` I'm down with doing it (in a separate PR maybe, but still).

File: README.md [hunk 1]
```diff
@@ -44,11 +44,18 @@ Somethings missing? Contributions to our docs are welcome!
 
 ## Support
 
-Ember-Electron is a small open source project. Use [GitHub Issues](https://github.com/felixrieseberg/ember-electron/issues) to report bugs and errors within the addon. 
+Ember-Electron is a small open source project. Use [GitHub Issues](https://github.com/felixrieseberg/ember-electron/issues) to report bugs and errors within the addon.
 
-If you need help *using* the addon with your application, feel free to drop by the [Ember Community Slack](https://emberjs.com/community/) and ask in `#e-electron`. If you have questions regarding Electron, their [Slack and forum](https://electron.atom.io/contact/) will be helpful as well. 
+If you need help *using* the addon with your application, feel free to drop by the [Ember Community Slack](https://emberjs.com/community/) and ask in `#e-electron`. If you have questions regarding Electron, their [Slack and forum](https://electron.atom.io/contact/) will be helpful as well.
```
Comment (jacobq) [inline README.md]: Ember community [now uses Discord](https://emberjs.com/community/), BTW (nothing personal, I'm sure).

**Comment by felixrieseberg** [comment]
Oh, got it! That sounds like a really good idea. Let me tackle that right after this PR 🙇

**Comment by jacobq** [comment]
Aha, here's the tool I think we want: https://github.com/marionebl/commitlint

**Comment by jacobq** [comment]
See also https://github.com/typed-ember/semantic-release-config
